### PR TITLE
refactor(frontend): always Open Tab Menu

### LINKED ISSUE #2532: 🔧 [Refactor] Top Menu shall never close
## 🔧 Refactor

Top Menu shall never close

The users do not find the cockpit, since its hidden in a collapsible menu.

It was decided by Hardi that the menu will never collapse, the animation between the menu entries stays as it is.

Undesired:
image

Desired:
image

### From #2507 

Menu-Design is infunctional.

All user tests show that cockpit is not found by the users. This happens because it is unclear that the menu is expandable.
Investigate if we need to change this.

### Todo

- Remove menu collapse
- ensure other transitions and animations stay as they are

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest

This PR modifies the TabControl component to keep the menu always open, addressing the issue of users not finding the cockpit option. The changes remove the auto-closing functionality while preserving the sliding animation between menu items.

### TabControl.vue
1. Removed `isOpen` ref and related logic
2. Removed `open()` and `closeWithDelay()` functions
3. Removed timer and related `onUnmounted` hook
4. Modified `setItem()` function to handle only the sliding animation
5. Simplified CSS by removing logic related to opening and closing
6. Kept `isSliding` state to manage the sliding animation
7. Removed click handler that was opening the menu
8. Adjusted CSS to always show all items and keep the menu in its open state

### TabControl.test.ts
1. Removed `vi.useFakeTimers()` and `vi.runAllTimers()` calls
2. Removed "click tab control button" describe block and nested tests
3. Removed "set item with menu closed" test
4. Removed "unmount" test checking for cleared timeouts
5. Added new test "always shows all items" to ensure all menu items are visible
6. Kept existing tests for setting active items based on routes and changing active item on click

## Impact
- The tab menu will now always remain open, improving discoverability of all menu items
- Sliding animation between menu items is preserved
- User experience is simplified by removing the collapsible behavior

## Testing
- Updated existing tests to reflect new always-open behavior
- Added new test to ensure all items are always visible
- Manually verified that the menu remains open and functions as expected

https://github.com/user-attachments/assets/7cd7db3f-00b5-4ef9-848d-519ea0fb041c

### Issues
- fixes #2532 

_________________________________

This PR description was generated with the assistance of an AI language model.

### CODE REVIEW

File: frontend/src/components/menu/TabControl.vue [hunk 1]
```diff
@@ -128,20 +79,15 @@ function setItem(item: number) {
   itemRef.addEventListener('transitionend', listener)
 
   requestAnimationFrame(() => {
-    // Move the marker to the active item
     moveMarker()
-
-    closeWithDelay()
+    setTimeout(() => {
+      isSliding.value = false
+    }, 500) // Match this with the transition time in CSS
```
Comment (ulfgebhardt) [inline frontend/src/components/menu/TabControl.vue:85]: this seems sketchy to me and requires a check by the frontend developers

File: frontend/src/components/menu/TabControl.vue [hunk 2]
```diff
@@ -1,10 +1,5 @@
 <template>
-  <button
-    ref="tabControl"
-    class="tab-control ma-auto border-sm"
-    :class="{ open: isOpen, sliding: isSliding }"
```
Comment (ulfgebhardt) [inline frontend/src/components/menu/TabControl.vue:5]: why is `sliding` removed?

**Review by sebastian2357 — CHANGES_REQUESTED** [review]
The PR changes the animation behavior of the item highlighting in the tab menu. Please compare the animation with the previous one.

OR let it be approved by Hardi.

File: frontend/src/components/menu/TabControl.vue [hunk 3]
```diff
@@ -128,20 +81,15 @@ function setItem(item: number) {
...
   })
 }
 
 onMounted(() => {
   moveMarker()
-  closeWithDelay()
-})
-
-onUnmounted(() => {
-  clearTimeout(timer)
```
Comment (sebastian2357) [inline frontend/src/components/menu/TabControl.vue:144]: Please ensure that all timers are get cleared when the component is getting to unmount

**Comment by sebastian2357** [comment]
In general I didn't check it in deep yet, but would it be possible to avoid the JS timers completely and only use CSS transitions? These are much easier to maintain.

**Comment by ogerly** [comment]
https://github.com/user-attachments/assets/9e7c2188-25ae-4b53-9cf4-83bdfc7fc2f6

**Pull request closed without merging**